daffodil-test/src/test/resources/test space/test 1/multi_base_05_nons.dfdl.xsd

```diff
@@ -28,8 +28,9 @@
         alignment="implicit" alignmentUnits="bytes" trailingSkip="0"/>
     </xs:appinfo>
   </xs:annotation>
-   
-  <xs:include schemaLocation="test space/test 2/multi_A_05_nons.dfdl.xsd"/>
+
+  <!-- schemaLocations can either be relative paths, or can be absolute paths -->
+  <xs:include schemaLocation="../../test space/test 2/multi_A_05_nons.dfdl.xsd"/>
```

Comment:
*This comment isn't about this line, but allows for threaded discussion*

Just to throw it out and spur discussion, another completely different option is to punt on the issue entirely with minor tweaks to resolution:

1. Relative imports (i.e. schemaLocations that do not start with a `/`) are looked up relative to the base and we never look at the classpath. If it fails to resolve relatively, the lookup fails. This works for files and intra-jar lookups and non-Daffodil resolvers that don't know about classpaths/jars since they already support relative lookups.
   
   Note that this breaks things like `schemaLocation="org/apache/daffodil/xsd/DFDLGenralFormat.dfdl.xsd"`, but maybe we make exceptions for schema locations that start with `org/apached/daffodil` and always look those up on the classpath, and warn that it should be changed to be absolute.
2. Absolute paths (i.e. schemaLocations that start with a `/`) are looked up absolutely on the classpath and if not found the looked up absolutely on the file system--no relative lookups. If a schema expects an include to be in another jar or directory on the classpath, an absolute path **must** be used. This works for jars and directories on the classpath, but **not** for non-Daffodil resolvers that don't use a classpath. In cases where those are used, pre-processing must be done to convert those absolute paths to paths that work in the system.
   
   And preprocessing could maybe be done with a one-liner bash command, e.g.
   ```bash
   find /path/to/schemas/root -type f -print0 | \
     xargs -0 sed -i 's@schemaLocation="/@schemaLocation="/path/to/schemas/root/@'
   ```

  This modified schemas should work for both Daffodil and non-Daffodil resolvers. A more complex script could alternatively calculate and prepend the right number of up-steps to these paths if the root changed.

This approach is very similar to what we have now, but makes the absolute path explicit. There's no guessing if a path should be on the classpath or not, it's either relative and looked up relatively, or it's not and it's looked up on the classpath. By being explicit about requiring the `/`, it is also more clear which schemaLocations must be modified if classpath resolution isn't supported--it's any schemaLocation that starts with a `/`. Additionally, if directory levels are added or removed in the including schema, the absolute paths still work--no need to add or remove up-steps to get to the root if dirs are moved around.

I imagine some systems that use non-Daffodil resolvers could even (or might already?) do something like put their applications in a chroot or mount namespace, in which case the absolute paths might actually work unmodified? That's probably much more work than just pre-processing the paths, but is a potential alternative, and maybe even a good idea for security relevant applications.

Reply:
Makes sense to me.  I like the simpler rules making relative paths always look at the file system (except for an org/apache/daffodil heuristic if necessary) and absolute paths always look at the classpath.